File: problems/Problem 002.py
# By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

import time

sum = 0
def sum_of_fibonacci(a,b,n):
  global sum
  if b <= n:
    if b%2 == 0:
      sum += b
    sum_of_fibonacci(b,a+b,n)
  else:
    print(sum)

def main():
  sum_of_fibonacci(1,2,4000000)

if __name__ == "__main__":
  start_time = time.perf_counter()
  main()
  print("Program executed in {r:1.4f} second(s)".format(r=time.perf_counter()-start_time))

File: problems/Problem 002.cpp
#include <iostream>
// By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

int ans = 0;

void sum_of_even_fib(int a, int b, int n){
  if(b <= n){
    if(b % 2 == 0){
      ans += b;
    }
    sum_of_even_fib(b,a+b,n);
  } 
}

int main(){
  int max_n;
  std::cout << "Sum of even-valued Fibonacci numbers under: ";
  std::cin >> max_n;
  sum_of_even_fib(1, 2, max_n);
  std::cout << ans << std::endl;
  return 0;
}